extension/src/prototypes/ui-components/TimelineParameterItem.tsx

```diff
@@ -0,0 +1,276 @@
+// Note: this component does not follow the pattern of the other parameterItem components.
+import PauseIcon from "@mui/icons-material/Pause";
+import PlayArrowIcon from "@mui/icons-material/PlayArrow";
+import { styled, IconButton, Select, Typography, SelectChangeEvent } from "@mui/material";
+import { useAtom } from "jotai";
+import { FC, useCallback, useEffect, useMemo, useRef, useState } from "react";
+
+import { activeTimelineComponentIdAtom } from "../../shared/view/state/timeline";
+
+import { SelectItem } from "./SelectItem";
+import TimelineBar from "./TimelineBar";
+
+type TimelineParameterItemProps = {
+  id: string;
+  start?: string;
+  current?: string;
+  end?: string;
+  timezone?: string;
+};
+
+const Timeline = styled("div")(() => ({
+  display: "flex",
+  flexDirection: "column",
+  gap: 8,
+  "*": {
+    boxSizing: "border-box",
+  },
+}));
+
+const Controls = styled("div")(({ theme }) => ({
+  display: "flex",
+  alignItems: "center",
+  justifyContent: "space-between",
+  gap: theme.spacing(1),
+}));
+
+const ButtonsWrapper = styled("div")(() => ({
+  display: "flex",
+  alignItems: "center",
+  justifyContent: "center",
+}));
+
+const ButtonWrapper = styled("div")(() => ({
+  width: 48,
+  height: 48,
+  display: "flex",
+  alignItems: "center",
+  justifyContent: "center",
+}));
+
+const StyledButton = styled(IconButton)<{ active?: number }>(({ active, theme }) => ({
+  minWidth: "unset",
+  padding: "0",
+  width: 36,
+  height: 36,
+  color: active ? theme.palette.primary.main : theme.palette.text.primary,
+}));
+
+const SelectWrapper = styled("div")(({ theme }) => ({
+  padding: theme.spacing(1, 0),
+}));
+
+const CurrentTime = styled("div")(({ theme }) => ({
+  display: "flex",
+  alignItems: "center",
+  justifyContent: "center",
+  height: 32,
+  color: theme.palette.text.secondary,
+  fontSize: theme.typography.body2.fontSize,
+}));
+
+const speedOptions = [
+  {
+    value: 60,
+    label: "1分 / 秒",
+  },
+  {
+    value: 1800,
+    label: "30分 / 秒",
+  },
+  {
+    value: 3600,
+    label: "1時間 / 秒",
+  },
+  {
+    value: 36000,
+    label: "10時間 / 秒",
+  },
+];
+
+export const TimelineParameterItem: FC<TimelineParameterItemProps> = ({
+  id,
+  start,
+  current,
+  end,
+  timezone = "+9",
+}) => {
+  const startDate = useMemo(() => new Date(start ?? ""), [start]);
+  const endDate = useMemo(() => new Date(end ?? ""), [end]);
+  const [currentDate, setCurrentDate] = useState(new Date(current ?? ""));
+
+  useEffect(() => {
+    setCurrentDate(new Date(current ?? ""));
+  }, [current]);
+
+  const [activeTimelineComponentId, setActiveTimelineComponentId] = useAtom(
+    activeTimelineComponentIdAtom,
+  );
```

Comment:
I think we should take `activeTimelineComponentIdAtom` from outside to keep this component being reusable.

Reply:
Updated!